### ISSUE TITLE
Training time way TOO LONG

### ISSUE THREAD

**Issue by Elderpotato** [matterport/Mask_RCNN#2388]
Hi, I want to check how much time do Mask RCNN usually take for training? Right now, I'm using 10 images (7 for training and 3 for validation) and this requires 3-4h per epoch and will take me more than 15h if I do 5 epochs. As I'm using Colab, it will probably get disconnected before training is completed. Also, the training time for such a small data seems ridiculous to me so I'm seeking advice here.

Additional info:
I'm using tensorflow 1.7 and keras 2.1.5 (the only combination I found to have work) since I realised the latest and other ver of TF seems to give me errors and I have spent a lot of time debugging it but to no avail. Image annotation was done in VIA and exported in json file. Initial trial seems to take me 5h per epoch so I tried to cropped and resize (downsize) as much as possible from ~2000x1500 to something like ~300x150. Even so, training time doesn't seem to have improved significantly, became 3h+ per epoch (might even be zero improvements if it's due to the fluctuation). Not all of my training images are of same size (not sure if it affects) and for each training image I have about 20-30 annotations of the object I want to detect. Also, I realised when running in Colab, I keep getting the warning message that I'm allocated a GPU but not actually using it which I don't understand why. Tried using tensorflow-gpu then I will get the same error as mentioned above and the code will not run, so I end up using tensorflow 1.7 again.

I'm very new to this field and would appreciate any advice on my current situation, or tips to speed up my training time. TIA!

**Comment by luna9722** [comment]
Hello, can i ask you how you managed to solve the problem ?

**Comment by FiyinfobaO** [comment]
> Hi, I want to check how much time do Mask RCNN usually take for training? Right now, I'm using 10 images (7 for training and 3 for validation) and this requires 3-4h per epoch and will take me more than 15h if I do 5 epochs. As I'm using Colab, it will probably get disconnected before training is completed. Also, the training time for such a small data seems ridiculous to me so I'm seeking advice here.
> 
> Additional info: I'm using tensorflow 1.7 and keras 2.1.5 (the only combination I found to have work) since I realised the latest and other ver of TF seems to give me errors and I have spent a lot of time debugging it but to no avail. Image annotation was done in VIA and exported in json file. Initial trial seems to take me 5h per epoch so I tried to cropped and resize (downsize) as much as possible from ~2000x1500 to something like ~300x150. Even so, training time doesn't seem to have improved significantly, became 3h+ per epoch (might even be zero improvements if it's due to the fluctuation). Not all of my training images are of same size (not sure if it affects) and for each training image I have about 20-30 annotations of the object I want to detect. Also, I realised when running in Colab, I keep getting the warning message that I'm allocated a GPU but not actually using it which I don't understand why. Tried using tensorflow-gpu then I will get the same error as mentioned above and the code will not run, so I end up using tensorflow 1.7 again.
> 
> I'm very new to this field and would appreciate any advice on my current situation, or tips to speed up my training time. TIA!

Hi, Were you able to solve the problem? I'm currently facing the same issue.
